src/test/scala/uk/gov/nationalarchives/draftmetadatavalidator/LambdaSpec.scala

```diff
@@ -24,41 +24,103 @@ class LambdaSpec extends ExternalServicesSpec {
     authOkJson()
     graphqlOkJson(true)
     mockS3GetResponse("sample.csv")
+    mockS3ErrorFilePutResponse()
     val input = Map("consignmentId" -> consignmentId).asJava
     val response = new Lambda().handleRequest(input, mockContext)
     response.getStatusCode should equal(200)
+
+    val s3Interactions: Iterable[ServeEvent] = wiremockS3.getAllServeEvents.asScala.filter(serveEvent => serveEvent.getRequest.getMethod == RequestMethod.PUT).toList
+    s3Interactions.size shouldBe 1
+
+    val errorWriteRequest = s3Interactions.head
+    val errorFileData = errorWriteRequest.getRequest.getBodyAsString
+    val today = org.joda.time.DateTime.now().toString("yyyy-MM-dd")
+    val expectedErrorData = s"""{
+                               |  "consignmentId" : "f82af3bf-b742-454c-9771-bfd6c5eae749",
+                               |  "date" : "$today",
+                               |  "validationErrors" : [
+                               |  ]
+                               |}""".stripMargin
+    errorFileData shouldBe expectedErrorData
   }
 
-  "handleRequest" should "download the draft metadata csv file, validate it and re-upload to s3 bucket if it has any errors" in {
+  "handleRequest" should "download the draft metadata csv file, validate it and save error file to s3" in {
     authOkJson()
     graphqlOkJson()
     mockS3GetResponse("invalid-sample.csv")
-    mockS3PutResponse()
+    mockS3ErrorFilePutResponse()
     val input = Map("consignmentId" -> consignmentId).asJava
     val response = new Lambda().handleRequest(input, mockContext)
     response.getStatusCode should equal(200)
 
     val s3Interactions: Iterable[ServeEvent] = wiremockS3.getAllServeEvents.asScala.filter(serveEvent => serveEvent.getRequest.getMethod == RequestMethod.PUT).toList
     s3Interactions.size shouldBe 1
 
-    val csvWriteEvent = s3Interactions.head
-    val expectedCSVHeader =
-      "Filename,Filepath,Date last modified,Closure status,Closure Start Date,Closure Period,FOI exemption code,FOI decision asserted,Is the title sensitive for the public?,Add alternative title without the file extension,Description,Is the description sensitive for the public?,Alternative description,Language,Date of the record,Translated title of record,Former reference,UUID,Error"
-    val expectedCSVRow1 =
-      "test3.txt,test/test3.txt,12/2/2345,Closed,,,,,No,,hhhhh,No,,English,,,,a060c57d-1639-4828-9a7a-67a7c64dbf6c,foi_exemption_asserted: type | foi_exemption_code: type | closure_period: type | closure_start_date: type"
-    val csvLines = csvWriteEvent.getRequest.getBodyAsString.split("\\n")
-    csvLines(0).strip() shouldBe expectedCSVHeader
-    csvLines(1).strip() shouldBe expectedCSVRow1
+    val errorWriteRequest = s3Interactions.head
+    val errorFileData = errorWriteRequest.getRequest.getBodyAsString
+    val today = org.joda.time.DateTime.now().toString("yyyy-MM-dd")
+    val expectedErrorData = s"""{
```

Comment:
This is should go as a Json file in the test resources to be read instead of hard coding into the test file.